hack/setup.sh

```diff
@@ -248,22 +339,106 @@ install_tekton() {
                 kapp deploy --yes -a tekton -f-
 }
 
-setup_example() {
+install_tekton_git_cli_task() {
+  kapp deploy --yes -a tekton-git-cli -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/git-cli/0.2/git-cli.yaml
+}
+
+setup_source_to_gitops() {
+        log "setting up source to gitops"
+
+        touch hack/git_entropy
+        echo $RANDOM | base64 > hack/git_entropy
+
+        GIT_WRITER_SERVER_PUBLIC_TOKEN=${GIT_WRITER_SERVER_PUBLIC_TOKEN:-"$(cat hack/gitea-server-public-key)"}
+
         ytt --ignore-unknown-comments \
-                -f "$DIR/../examples/source-to-knative-service" \
+                -f "$DIR/../examples/source-to-gitops" \
                 --data-value registry.server="$REGISTRY" \
                 --data-value registry.username=admin \
                 --data-value registry.password=admin \
-                --data-value image_prefix="$REGISTRY/example-" |
-                kapp deploy --yes -a example -f-
+                --data-value image_prefix="$REGISTRY/example-" \
+                --data-value git_writer.message="Some peturbation: $(cat hack/git_entropy)" \
+                --data-value git_writer.ssh_user="$GIT_WRITER_SSH_USER" \
+                --data-value git_writer.username="$GIT_WRITER_USERNAME" \
+                --data-value git_writer.user_email="$GIT_WRITER_SSH_USER_EMAIL" \
+                --data-value git_writer.server="$GIT_WRITER_SERVER" \
+                --data-value git_writer.port="$GIT_WRITER_SERVER_PORT" \
+                --data-value git_writer.repository="$GIT_WRITER_PROJECT/$GIT_WRITER_REPOSITORY.git" \
+                --data-value git_writer.branch="$(cat hack/git_entropy)" \
+                --data-value git_writer.base64_encoded_ssh_key="$(echo "$(cat hack/gitea-key)" | base64)" \
+                --data-value git_writer.base64_encoded_known_hosts="$(echo "$GIT_WRITER_SERVER_PUBLIC_TOKEN" | base64)" |
+                kapp deploy --yes -a example-supply -f-
+}
+
+setup_gitops_to_app() {
+        log "setting up gitops to app"
+
+        GIT_WRITER_SERVER_PUBLIC_TOKEN=${GIT_WRITER_SERVER_PUBLIC_TOKEN:-"$(cat hack/gitea-server-public-key)"}
+
+        ytt --ignore-unknown-comments \
+                -f "$DIR/../examples/gitops-to-app" \
+                --data-value git_writer.server="$GIT_WRITER_SERVER:$GIT_WRITER_SERVER_PORT" \
+                --data-value git_writer.repository="$GIT_WRITER_PROJECT/$GIT_WRITER_REPOSITORY" \
+                --data-value git_writer.branch="$(cat hack/git_entropy)" \
+                --data-value git_writer.base64_encoded_ssh_key="$(echo "$(cat hack/gitea-key)" | base64)" \
+                --data-value git_writer.base64_encoded_known_hosts="$(echo "$GIT_WRITER_SERVER_PUBLIC_TOKEN" | base64)" |
+                kapp deploy --yes -a example-deliver -f-
 }
 
 teardown_example() {
-        kapp delete --yes -a example
+        kapp delete --yes -a example-supply
+        kapp delete --yes -a example-deliver
+        rm hack/git_entropy
```

Comment:
```suggestion
        rm hack/git_entropy || true
```
so that it doesn't fail if you don't have the file